pkg/overlay/cache.go

```diff
@@ -220,3 +231,47 @@ func randomID() ([]byte, error) {
 	_, err := rand.Read(result)
 	return result, err
 }
+
+// getNodeRep gets a node's stats from statdb and returns a node with reputation attached
+func (o *Cache) getNodeRep(ctx context.Context, n *pb.Node) (*pb.Node, error) {
+	stats, err := o.statdb.CreateEntryIfNotExists(ctx, node.IDFromString(n.Id).Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	n.Reputation = &pb.NodeRep{
+		UptimeRatio:       stats.UptimeRatio,
+		AuditSuccessRatio: stats.AuditSuccessRatio,
+		AuditCount:        stats.AuditCount,
```

Comment:
are all of these stats saved when the node is createdIfNotExists too? or is it not necessary to save them at that time?